src/Microsoft.Diagnostics.Monitoring.StartupHook/Exceptions/Pipeline/Steps/ExceptionEventsPipelineStep.cs

```diff
@@ -105,5 +110,56 @@ private static ReadOnlySpan<StackFrame> ComputeEffectiveCallStack(Exception exce
             }
             return ReadOnlySpan<StackFrame>.Empty;
         }
+
+        private ulong GetExceptionId(Exception? exception)
+        {
+            if (null == exception)
+                return 0;
+
+            if (TryGetExceptionId(exception, out ulong exceptionId))
+                return exceptionId;
+
+            lock (exception.Data)
+            {
+                if (TryGetExceptionId(exception, out exceptionId))
+                    return exceptionId;
+
+                exceptionId = Interlocked.Increment(ref _nextExceptionId);
+
+                exception.Data[ExceptionIdKey] = exceptionId;
+            }
+
+            return exceptionId;
+
+            static bool TryGetExceptionId(Exception exception, out ulong exceptionId)
+            {
+                if (exception.Data.Contains(ExceptionIdKey))
+                {
+                    // The ExceptionIdKey data should only ever have a ulong
+                    if (exception.Data[ExceptionIdKey] is ulong exceptionIdCandidate)
+                    {
+                        exceptionId = exceptionIdCandidate;
+                        return false;
```

Comment:
I'm assuming this should probably be true?

Reply:
(smacks head) yes